azavea/fb-gender-survey-dashboard
# Establish initial data processing pipeline

## Overview

The data processing for this project will be modest, but we can anticipate periodic data updates. Include an environment to isolate dependencies and process the source Excel files into a JSON format expected by the application. The end result files generated by this PR are not representative of the actual data expected, but demonstrate the end-to-end functionality of the system.

To confirm the process works as expected, the example data generated is processed and added to the application, which has been altered to show that the data is loaded.

Connects #3 

### Notes

This only partially completes #3. The actual data processing will be implemented when the final source is release, but should slot into this framework and be more easily reviewed in isolation.

### Demo

![image](https://user-images.githubusercontent.com/1014341/100773697-3972be00-33cf-11eb-999c-582a6def9266.png)


## Testing Instructions

 * Follow the instructions in the updated readme to generate the app data
 * Ensure that the two datafiles appear in src/dataproc/output
 * Ensure that the git working directory is clean, these files should be not checked into the repository

 * Use the deploy preview to confirm that the data files included in the app are accessible to the application in the hosting environment.
